evennia/contrib/game_systems/achievements/achievements.py

```diff
@@ -0,0 +1,370 @@
+"""
+Achievements
+
+This provides a system for adding and tracking player achievements in your game.
+
+Achievements are defined as dicts, loosely similar to the prototypes system.
+
+An example of an achievement dict:
+
+    EXAMPLE_ACHIEVEMENT = {
+        "name": "Some Achievement",
+        "desc": "This is not a real achievement.",
+        "category": "crafting",
+        "tracking": "box",
+        "count": 5,
+        "prereqs": "ANOTHER_ACHIEVEMENT",
+    }
+
+The recognized fields for an achievement are:
+
+- name (str): The name of the achievement. This is not the key and does not need to be unique.
+- desc (str): The longer description of the achievement. Common uses for this would be flavor text
+        or hints on how to complete it.
+- category (str): The type of things this achievement tracks. e.g. visiting 10 locations might have
+        a category of "post move", or killing 10 rats might have a category of "defeat".
+- tracking (str or list): The *specific* thing this achievement tracks. e.g. the above example of
+        10 rats, the tracking field would be "rat".
+- tracking_type: The options here are "sum" and "separate". "sum" means that matching any tracked
+        item will increase the total. "separate" means all tracked items are counted individually.
+        This is only useful when tracking is a list. The default is "sum".
+- count (int): The total tallies the tracked item needs for this to be completed. e.g. for the rats
+        example, it would be 10. The default is 1
+- prereqs (str or list): An optional achievement key or list of keys that must be completed before
+        this achievement is available.
+
+To add achievement tracking, put `track_achievements` in your relevant hooks.
+
+Example:
+
+    def at_use(self, user, **kwargs):
+        # track this use for any achievements about using an object named our name
+        finished_achievements = track_achievements(user, category="use", tracking=self.key)
```

Comment:
Why not use the actual achievement-dict you defined above as the example when calling this function? This example doesn't make it clear to me how this function interacts with the achievement.

Reply:
This question is still open to me.

Reply:
The code reviewed here is outdated. The updated docstring resolved this.